### DISCUSSION TITLE
CSLA0003 and No-Argument Public Constructors

### DISCUSSION THREAD

**Discussion by JasonBock** [MarimerLLC/csla#1742] [General]
This is something that I didn't consider:
```
public sealed class Customer
  : BusinessBase<Customer>
{
  // Operations, properties, etc. go here...
}
```
My point is, this code doesn't explicitly define a no-argument public constructor, which will trip the CSLA0003 analyzer. But, I believe the C# compiler will automatically add this constructor in if you haven't defined *any* constructors in a class. So I think I might need to be a bit more cautious with this analyzer. If you have no constructors defined, you should be OK. Thoughts?

**Comment by rockfordlhotka** [comment]
I agree. Defining no ctor at all means C# adds a default ctor, and that meets the requirement for CSLA serialization.

**Comment by JasonBock** [comment]
Never mind :). I just wrote this:
```
[Serializable]
public sealed class Customer
  : BusinessBase<Customer>
{ }
```
And I got no analyzer errors. So I must be handling this condition.
